backends-clickhouse/src/main/scala/org/apache/spark/sql/execution/datasources/utils/MergeTreePartsPartitionsUtil.scala

```diff
@@ -55,6 +55,9 @@ object MergeTreePartsPartitionsUtil extends Logging {
     }
     val fileIndex = relation.location.asInstanceOf[TahoeFileIndex]
 
+    // when querying, use deltaLog.update(true) to get the staleness acceptable snapshot
+    val latestCommitTimestamp = table.deltaLog.update(true).timestamp
```

Comment:
will this impact perf benchmark ? need perf result

Reply:
set `spark.databricks.delta.stalenessLimit` to non-zero, it will launch a thread to update the snapshot:
![image](https://github.com/apache/incubator-gluten/assets/9430290/21dd709b-ca3a-4c4f-92eb-74793a17fc26)

Reply:
Merge this pr first, will test perf later